packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -174,29 +225,27 @@ export class PersonalizeMiddleware extends MiddlewareBase {
       return response;
     }
 
-    if (!browserId) {
-      browserId = await this.cdpService.generateBrowserId();
-
-      if (!browserId) {
-        debug.personalize('skipped (browser id generation failed)');
-        return response;
-      }
-    }
-
-    // Execute targeted experience in CDP
-    const { ua } = userAgent(req);
     const params = this.getExperienceParams(req);
-    const pointOfSale = this.config.getPointOfSale
-      ? this.config.getPointOfSale(site, language)
-      : PosResolver.resolve(site, language);
-    const variantId = await this.cdpService.executeExperience(
+    const { ua } = userAgent(req);
+
+    debug.personalize(
+      'executing experience for %s %s %s %o',
       personalizeInfo.contentId,
-      browserId,
       ua,
-      pointOfSale,
       params
     );
 
+    // Execute targeted experience in CDP
+    const personalizationData = {
+      channel: 'WEB',
+      currency: 'USA',
```

Comment:
This should probably be added to our config (like `channel`) and fallback to 'USA'?

Reply:
You mean add it as a environment variable?

Reply:
No need for env variable, just make it configurable for the middleware by adding it as a prop on `CdpServiceConfig` (like `channel`). Use that if present and fallback to 'USA' here.
```suggestion
      currency: this.config.cdpConfig.currency ?? 'USA',
```